server/registry/actions_artifacts_test.go

```diff
@@ -206,6 +206,46 @@ func TestCreateArtifactResponseCodes(t *testing.T) {
 			},
 			want: codes.NotFound,
 		},
+		{
+			desc: "parent api not found",
+			seed: &rpc.Project{Name: "projects/my-project"},
+			req: &rpc.CreateArtifactRequest{
+				Parent:     "projects/other-project/locations/global/apis/a",
+				ArtifactId: "valid-id",
+				Artifact:   &rpc.Artifact{},
+			},
+			want: codes.NotFound,
+		},
+		{
+			desc: "parent version not found",
+			seed: &rpc.Project{Name: "projects/my-project"},
+			req: &rpc.CreateArtifactRequest{
+				Parent:     "projects/other-project/locations/global/apis/a/versions/v",
```

Comment:
In this test case, the api `a` itself doesn't exist. For me, it seems more like an `API not found` test case than `version not found`. 

Same for below cases.

Reply:
That's true - the description is just a label here, and the returned error code is a general "NotFound" that doesn't specify what is missing. The implementation only verifies that the parent exists. If no parent exists, the CreateArtifact implementation might climb the ancestry to report the highest-level missing ancestor, but currently it doesn't do that.